test/jdk/sun/net/www/http/KeepAliveStream/KeepAliveStreamCloseWithWrongContentLength.java

```diff
@@ -103,7 +136,7 @@ public static void main (String[] args) throws Exception {
             }
             is.close();
         } catch (IOException e) {
-            return;
+            throw new RuntimeException (e);
         } catch (NullPointerException e) {
             throw new RuntimeException (e);
         } finally {
```

Comment:
The finally block should also do:

```
var socket = server.socket;
if (socket != null) socket.close();
```

Or better add a close() method on XServer that would both close socket and serverSocket - and make XServer AutoCloseable so that you can transform your try { } finally {} into a try-with-resource.